.github/workflows/druid-test-image.yml

```diff
@@ -0,0 +1,36 @@
+name: Build Druid image for tests
+
+on:
+  pull_request:
+    paths:
+      - "scripts/test-druid-micro.dockerfile"
+      - ".github/workflows/druid-test-image.yml"
```

Comment:
Should only run on merge to main, something like:
```
on:
  push:
    branch:
      - main
    paths:
      - ...
```

Reply:
That way if one changes tests that require image update then it will require 2 separate PRs (one for the Dockerfile to be merged and the other for test changes). Instead if you build on push to PR the changes can be packaged in a single PR.